### PR TITLE
feat: share qr code on payment request

### LINKED ISSUE #579: add QR code when the user is doing a money request

### PR DESCRIPTION
**Pull request by rhbrunetto**
## Changes

- Adds QRCode to be shared on Payment Request

<details><summary>Screenshots iPhone 14 Pro Max</summary

Simulator Screen Shot - iPhone 14 Pro Max - 2022-10-31 at 10 22 41

Simulator Screen Shot - iPhone 14 Pro Max - 2022-10-31 at 10 22 55

</details>

<details><summary>Screenshots iPhone SE</summary

Simulator Screen Shot - iPhone SE (3rd generation) - 2022-10-31 at 10 18 45

Simulator Screen Shot - iPhone SE (3rd generation) - 2022-10-31 at 10 19 09

</details>

## Related issues

Fixes #579 

## Checklist

- [x] PR is ready for review (if not, it should be a draft).
- [x] PR title follows [Conventional Commits][1] guidelines.
- [x] Screenshots/video added.
- [ ] Tests added.
- [x] Self-review done.

[1]: https://www.conventionalcommits.org/en/v1.0.0/

### CODE REVIEW

File: packages/cryptoplease/lib/app/screens/authenticated/wallet_flow/wallet_flow_screen.dart [hunk 1]
```diff
@@ -136,3 +154,5 @@ extension on _Operation {
     }
   }
 }
+
+extension on SolanaPayRequest {}
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/app/screens/authenticated/wallet_flow/wallet_flow_screen.dart:158]: Empty code.

File: packages/cryptoplease/lib/ui/tab_bar.dart [hunk 2]
```diff
@@ -2,30 +2,73 @@ import 'package:flutter/material.dart';
...
+  }
+
+  Color get _unselectedLabelColor {
+    switch (variant) {
+      case CpTabBarVariant.dark:
+        return CpColors.disabledTabColor;
+      case CpTabBarVariant.inverted:
+        return Colors.black;
+    }
+  }
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/ui/tab_bar.dart:51]: They can be made file-private.

File: packages/cryptoplease/lib/features/payment_request/src/presentation/link_details/share_request.dart [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import '../../../../../ui/content_padding.dart';
+import '../../../../../ui/rounded_rectangle.dart';
+import '../../../../../ui/share_message/share_message_bubble.dart';
+import '../../../../../ui/share_message/share_message_header.dart';
+import '../../../../../ui/tab_bar.dart';
+import '../../../../../ui/theme.dart';
+import '../../bl/payment_request.dart';
+
+part 'components/share_link.dart';
+part 'components/share_qr_code.dart';
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/features/payment_request/src/presentation/link_details/share_request.dart:24]: Why `part`? They don't use any private API, and are even located in a different folder.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

File: packages/cryptoplease/lib/ui/tab_bar.dart (hunk 2 above)
Comment (ookami-kb) [reply packages/cryptoplease/lib/ui/tab_bar.dart:51]: @rhbrunetto this is not resolved.

Comment (rhbrunetto) [reply packages/cryptoplease/lib/ui/tab_bar.dart:51]: They were using the `variant` property of the class to retrieve the color.
I've extracted them from the class in the latest commit and passed the variant as a parameter.

Please let me know if it's not properly fixed.

**Review by ookami-kb — CHANGES_REQUESTED** [review]
1. Label shouldn't be shared in the link. Profile user name doesn't necessarily mean that the user is ok with sharing it. Until we have clear guidance on that, it's better not to display anything
2. QR-code should contain exactly the same link that is shared in the other tab, so that even without the app installed the payment can be initiated.

**Comment by antoineherzog** [comment]
for 2. I would be in favor to maintain Solana Pay comptability with others Solana Pay wallet. I dont expect a user to install the app by scanning a Resquest Payment QR code @ookami-kb

**Review by ookami-kb — APPROVED** [review]

**Pull request merged**